### PR TITLE
poddefaults: Extend tests for using images of each PR

### PR DESCRIPTION
**Pull request by apo-ger**
This PR resolves part of: https://github.com/kubeflow/kubeflow/issues/6766

Changes: 
-  Publish the docker image, only when a PR is merged
-  Remove  kind and manifests GH action tests
-  Introduce a GH action intergration test to ensure that on each PR commit:
   - The docker image can be built
   - The manifests can be built
   - The manifests with the updated image can be applied in a KinD cluster
   - The Pods become Ready

### CODE REVIEW

File: .github/workflows/poddefaults_intergration_test.yaml [hunk 1]
```diff
@@ -26,7 +42,7 @@ jobs:
     - name: Install cert-manager
       run: |
         echo "Installing cert-manager ..."
-        CERT_MANAGER_VERSION="v1.5.0"
+        CERT_MANAGER_VERSION="v1.10.1"
```
Comment (kimwnasptd) [inline .github/workflows/poddefaults_intergration_test.yaml:45]: @apo-ger could you convert this script into a `.sh` file and place it under https://github.com/kubeflow/kubeflow/tree/master/components/testing/gh-actions?

We might need to reuse this for other components, i.e. https://github.com/kubeflow/kubeflow/pull/6734

Comment (apo-ger) [reply .github/workflows/poddefaults_intergration_test.yaml]: done!

File: components/testing/gh-actions/install_cert_manager.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+echo "Installing cert-manager ..."
+CERT_MANAGER_VERSION="v1.10.1"
+mkdir cert_tmp
+pushd cert_tmp > /dev/null
+    OS=$(go env GOOS); ARCH=$(go env GOARCH); \
+        curl -L -o cmctl.tar.gz https://github.com/cert-manager/cert-manager/releases/download/$CERT_MANAGER_VERSION/cmctl-$OS-$ARCH.tar.gz
+    tar xzf cmctl.tar.gz
+    sudo mv cmctl /usr/local/bin
+    cmctl x install
+popd
```
[the remark this answers is not in the document]
Comment (apo-ger) [reply components/testing/gh-actions/install_cert_manager.sh]: fixed

**Comment by kimwnasptd** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**